src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -187,9 +187,27 @@ const std::vector<std::shared_ptr<OperatorTask>>& SQLPipelineStatement::get_task
   return _tasks;
 }
 
-const std::shared_ptr<const Table>& SQLPipelineStatement::get_result_table() {
+const std::pair<SQLPipelineStatus, std::shared_ptr<const Table>&> SQLPipelineStatement::get_result_table() {
+  // Returns true if a transaction was set and that transaction was rolled back.
+  const auto was_rolled_back = [&]() {
```

Comment:
Wow, this lambda is doing too much.

Should be 


```c++
 DebugAssert(_transaction_context->phase() == TransactionPhase::Active ||
                      _transaction_context->phase() == TransactionPhase::RolledBack ||
                      _transaction_context->phase() == TransactionPhase::Committed,
                  "Transaction found in unexpected state");
if (_transaction_context->phase() == TransactionPhase::RolledBack) {
    return {SQLPipelineStatus::RolledBack, _result_table};
}
```

Reply:
Not sure what you are suggesting. You are missing the `if (_transaction_context)` part - where should that go?

Reply:
Not sure what you are suggesting. You are missing the `if (_transaction_context)` part - where should that go?

Reply:
I think I mostly take issue with lambda called "was_rolled_back" - which should just be `_transaction_context->phase() == TransactionPhase::RolledBack` doing more than that. Minor though, leave it as it is if you like it that way

Reply:
To identify a roll back, you need two steps: First, check if we have a context at all, and second, check if it was rolled back. I wouldn't consider the assertion when deciding if the lambda does "too much", as for me assertions are not "doing" anything. I can't come up with a better solution that doesn't duplicate code. If you have a better name, I'll be happy to use it.